feature: define blog slug in frontmatter

jkornblum opened the pull request:
Modify `gatsby-node.js` so that it reads in frontmatter slug and uses it for blog URL. If frontmatter is not defined or slug is not defined it will default to file path for slug (current behavior).

E.g. on the demo the first blog post is `demo.com/blog/2017-04-18--welcoming/` by adding `slug` to front matter like
```
---
title: 'Article #2'
tags:
  - test
slug: 2017-04-18/my-article
---
```
the new url would be `demo.com/blog/2017-04-18/my-article`

fabien0102 commented on gatsby-node.js:
```diff
@@ -36,9 +36,10 @@ exports.onCreateNode = ({node, boundActionCreators, getNode}) => {
   let slug;
   switch (node.internal.type) {
     case `MarkdownRemark`:
+      const fmSlug = (node.frontmatter && node.frontmatter.slug) ? node.frontmatter.slug : false
```
Here you can use `get` from lodash (https://lodash.com/docs/4.17.5#get), and I prefer long variable name instead of acronyms :wink:

```js
const frontmatterSlug = get(node, 'frontmatter.slug');
```

fabien0102 commented:
@jkornblum carefull you miss a semi-column ^^

```
gatsby-node.js:39:97
  ✖  39:97  Missing semicolon.
```

You can run `npm run lint:js` to be sure (normally it's automaticly done in a git hook with husky) 😉

jkornblum commented:
Thanks for feedback. I tried the one liner without the checks! I was burned 😄 

Anyways I updated and cleaned up code, added unit test, and changed one markdown file use the frontmatter slug.

Let me know if it looks okay.

fabien0102 commented:
It looks good, I just need to check why netlify don't want to deploy ^^
